subprojects/testing-native/src/main/java/org/gradle/nativeplatform/test/googletest/plugins/GoogleUnitTestPlugin.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.nativeplatform.test.googletest.plugins;
+
+import org.gradle.api.Action;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.file.DirectoryProperty;
+import org.gradle.api.tasks.TaskContainer;
+import org.gradle.language.base.plugins.LifecycleBasePlugin;
+import org.gradle.nativeplatform.tasks.InstallExecutable;
+import org.gradle.nativeplatform.test.cpp.plugins.CppUnitTestBasePlugin;
+import org.gradle.nativeplatform.test.tasks.RunTestExecutable;
+
+public class GoogleUnitTestPlugin implements Plugin<Project> {
```

Comment:
This doesn't really have anything to do with google test. We just want a generic 'run this executable' plugin. Then, later, we would add a sibling to this plugin with better integration for google test.

Reply:
Yep, there was another commit after this that Gary and I worked on after you left that created a `GoogleTest` task and scraped the output from google test execution.

I pushed that experiment to a branch and detangle it from here, so we just have a `cpp-unit-test` plugin that adds a `RunTestExecutable` and wires everything together.  We can use this to make a native-sample to use googleTest or catch or whatever with this.